.changes/next-release/bugfix-AmazonS3-16d0bb1.json

```diff
@@ -0,0 +1,6 @@
+{
+    "category": "Amazon S3", 
+    "contributor": "", 
+    "type": "bugfix", 
+    "description": "Fixes double checksum validation for GetObject. Now when ChecksumMode is enabled, x-amz-te:append-md5 will not be sent, and only the flexible checksum will be validated. If ChecksumMode is enabled and no ChecksumAlgorithm was specified when putting the object, no validation will be performed."
```

Comment:
>If ChecksumMode is enabled and no ChecksumAlgorithm was specified

Do you mean no ChecksumAlgorithm was returned?

Reply:
Can we update javadoc to clarify the behavior? https://github.com/aws/aws-sdk-java-v2/blob/master/services/s3/src/main/java/software/amazon/awssdk/services/s3/S3Configuration.java#L284

More specially, under what situations, the SDK will enable/validate checksum for put and get

Reply:
updated changelog and javadocs